systems/data-plan/package/pkg/server/package.go

```diff
@@ -0,0 +1,136 @@
+package server
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+	"github.com/ukama/ukama/systems/common/grpc"
+	pb "github.com/ukama/ukama/systems/data-plan/package/pb"
+	validations "github.com/ukama/ukama/systems/data-plan/package/pkg/validations"
+
+	"github.com/ukama/ukama/systems/data-plan/package/pkg/db"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type PackageServer struct {
+	packageRepo db.PackageRepo
+	pb.UnimplementedPackagesServiceServer
+}
+
+func NewPackageServer(packageRepo db.PackageRepo) *PackageServer {
+	return &PackageServer{packageRepo: packageRepo}
+}
+
+func (p *PackageServer) Get(ctx context.Context, req *pb.GetPackagesRequest) (*pb.GetPackagesResponse, error) {
+	logrus.Infof("GetPackages : %v  ,%v", req.GetOrgId(), req.GetId())
+
+	if validations.IsInvalidId(req.GetOrgId()) {
+		return nil, status.Errorf(codes.InvalidArgument, "OrgId is required.")
+	}
+
+	packages, err := p.packageRepo.Get(req.GetOrgId(), req.GetId())
+	if err != nil {
+		logrus.Error("error while getting package" + err.Error())
+		return nil, grpc.SqlErrorToGrpc(err, "packages")
+	}
+
+	packageList := &pb.GetPackagesResponse{
+		Packages: dbpackagesToPbPackages(packages),
+	}
+
+	return packageList, nil
+}
+
+func (p *PackageServer) Add(ctx context.Context, req *pb.AddPackageRequest) (*pb.AddPackageResponse, error) {
+	logrus.Infof("Add Package Name: %v, SimType: %v, Active: %v, Duration: %v, SmsVolume: %v, DataVolume: %v, Voice_volume: %v", req.Name, req.SimType, req.Active, req.Duration, req.SmsVolume, req.DataVolume, req.VoiceVolume)
+	_package := &db.Package{
+		Name:         req.GetName(),
+		Sim_type:     req.GetSimType().String(),
+		Org_id:       uint(req.GetOrgId()),
+		Active:       req.Active,
+		Duration:     uint(req.GetDuration()),
+		Sms_volume:   uint(req.GetSmsVolume()),
+		Data_volume:  uint(req.GetDataVolume()),
+		Voice_volume: uint(req.GetVoiceVolume()),
+		Org_rates_id: uint(req.GetOrgRatesId()),
+	}
+	err := p.packageRepo.Add(_package)
+	if err != nil {
+
+		logrus.Error("Error adding a package. " + err.Error())
+
+		return nil, status.Errorf(codes.Internal, "error adding a package")
+	}
+
+	return &pb.AddPackageResponse{Package: dbPackageToPbPackages(_package)}, nil
+
+}
+
+func (p *PackageServer) Delete(ctx context.Context, req *pb.DeletePackageRequest) (*pb.DeletePackageResponse, error) {
+	logrus.Infof("Delete Packages, orgId: %v, packageId: %v", req.GetOrgId(), req.GetId())
+	if validations.IsInvalidId(req.GetOrgId()) {
```

Comment:
we also need to check these validations against those happening at the handler level.